src/python/zquantum/core/utils.py

```diff
@@ -251,10 +251,14 @@ def to_dict(self):
         """Convert to a dictionary"""
 
         data = {'schema' : SCHEMA_VERSION + '-value_estimate'}
- 
-        data['value'] = self.value
+        if type(self.value).__module__ == np.__name__:
+            data['value'] = self.value.item()
+        else:
+            data['value'] = self.value
 
-        if self.precision:
+        if type(self.precision).__module__ == np.__name__:
+            data['precision'] = self.precision.item()
+        else:
```

Comment:
Can you explain what's going on here?

Reply:
This is for casting from `np.float64` to python `float`. 
Cause `np.float64` is non-serializable so it caused problems.
I agree this doesn't look well, but it's the best way I found.